server/service/src/requisition_line/response_requisition_line/insert/generate.rs

```diff
@@ -1,36 +1,62 @@
-use repository::{RequisitionLineRow, RequisitionRow};
+use chrono::Utc;
+use repository::{ItemRow, RequisitionLineRow, RequisitionRow};
 
-use crate::{
-    requisition::request_requisition::generate_requisition_lines, service_provider::ServiceContext,
+use crate::requisition::request_requisition::{
+    generate_suggested_quantity, GenerateSuggestedQuantity,
 };
 
-use super::{InsertResponseRequisitionLine, OutError};
+use super::InsertResponseRequisitionLine;
 
 pub fn generate(
-    ctx: &ServiceContext,
-    store_id: &str,
     requisition_row: RequisitionRow,
+    item_row: ItemRow,
     InsertResponseRequisitionLine {
         id,
         requisition_id: _,
-        item_id,
-        their_stock_on_hand,
-        requested_quantity,
+        item_id: _,
         supply_quantity,
         comment,
+        stock_on_hand,
+        requested_quantity,
+        average_monthly_consumption,
+        incoming_units,
+        outgoing_units,
+        loss_in_units,
+        addition_in_units,
+        expiring_units,
+        days_out_of_stock,
+        option_id,
     }: InsertResponseRequisitionLine,
-) -> Result<RequisitionLineRow, OutError> {
-    let mut new_requisition_line =
-        generate_requisition_lines(ctx, store_id, &requisition_row, vec![item_id])?
-            .pop()
-            .ok_or(OutError::CannotFindItemStatusForRequisitionLine)?;
+) -> RequisitionLineRow {
+    let suggested_quantity = generate_suggested_quantity(GenerateSuggestedQuantity {
+        average_monthly_consumption: average_monthly_consumption.unwrap_or(0.0),
+        available_stock_on_hand: stock_on_hand.unwrap_or(0.0),
+        min_months_of_stock: requisition_row.min_months_of_stock,
+        max_months_of_stock: requisition_row.max_months_of_stock,
+    });
 
-    new_requisition_line.id = id;
-    new_requisition_line.initial_stock_on_hand_units = their_stock_on_hand.unwrap_or(0.0);
-    new_requisition_line.available_stock_on_hand = their_stock_on_hand.unwrap_or(0.0);
-    new_requisition_line.supply_quantity = supply_quantity.unwrap_or(0.0);
-    new_requisition_line.requested_quantity = requested_quantity.unwrap_or(0.0);
-    new_requisition_line.comment = comment.or(new_requisition_line.comment);
-
-    Ok(new_requisition_line)
+    RequisitionLineRow {
```

Comment:
Was using wrong method to generate the new requisition line on insert... should just be all from input since this method should only be used for inserting manual requisitions